packages/components/src/tag/src/Tag.tsx

```diff
@@ -48,29 +54,43 @@ const embedIconButton = createEmbeddableAdapter({
 });
 /* eslint-enable sort-keys, sort-keys-fix/sort-keys-fix */
 
-export function InnerTag({
-    active,
-    as = DefaultElement,
-    children,
-    disabled,
-    fluid,
-    focus,
-    forwardedRef,
-    hover,
-    onRemove,
-    size,
-    variant = "solid",
-    ...rest
-}: InnerTagProps) {
+export function InnerTag(props: InnerTagProps) {
+    const [fieldProps] = useFieldInputProps();
+
+    const {
+        active,
+        as = DefaultElement,
+        children,
+        disabled,
+        fluid,
+        focus,
+        forwardedRef,
+        hover,
+        onRemove,
+        size,
+        validationState,
+        variant = "solid",
+        ...rest
+    } = mergeProps(
+        props,
+        fieldProps
+    );
+    
     const fluidValue = useResponsiveValue(fluid);
     const sizeValue = useResponsiveValue(size);
 
     const ref = useMergedRefs(forwardedRef);
 
-    const { counter, dot, "end-icon": endIcon, icon, text } = useSlots(children, useMemo(() => ({
+    const { avatar, counter, dot, "end-icon": endIcon, icon, text } = useSlots(children, useMemo(() => ({
         _: {
             defaultWrapper: Text
         },
+        avatar: () => {
+            return {
+                className: "o-ui-tag-avatar",
+                size: "xs"
+            };
```

Comment:
i don't think this has to be a function, you could do something like : 

```
avatar: {
                className: "o-ui-tag-avatar",
                size: "xs"
            }
 
```

Reply:
true, it had parameters before, but I didn't need them.